core/src/main/java/org/testcontainers/containers/GenericContainer.java

```diff
@@ -575,18 +535,33 @@ public Integer getMappedPort(final int originalPort) {
         }
     }
 
+    /**
+     * <b>Resolve</b> Docker image and set it.
+     *
+     * @param dockerImageName image name
+     */
     public void setDockerImageName(@NonNull String dockerImageName) {
         DockerImageName.validate(dockerImageName);
-        this.dockerImageName = dockerImageName;
 
-        Profiler profiler = new Profiler("Rule creation - prefetch image");
-        profiler.setLogger(logger());
+        String[] parts = dockerImageName.split(":");
```

Comment:
ok. It's guarded by `DockerImageName.validate(dockerImageName);`, but ok, I'm fine to change it :) Will do it now